src/pages/credit.vue

```diff
@@ -0,0 +1,297 @@
+    };
+
+    /** credit-filter */
+    const mode = ref<CreditFilterMode>("filtering");
+    const yearOptions: string[] = [
+      "すべての年度",
+      "2022年度",
+      "2021年度",
+      "2020年度",
+      "2019年度",
+    ];
+    const selectedYear = ref(yearOptions[0]);
+    const totalCredit = ref(getTotalCredit(apiCourses.value));
+    const selectedTagId = ref<string | undefined>(undefined);
+    const creditTags = reactive<CreditTag[]>(
+      apiToCreditTags(apiCourses.value, apiTags.value)
+    );
+
+    const onCreditFilterCreateTag = (name: string) => {
+      // TODO: api につなぐ
+      tagRepo.createTag({ name, order: creditTags.length });
+
+      updateApiTags();
+      updateView();
+    };
+    const onCreditFilterUpdateTagName = (id: string, name: string) => {
+      // TODO: api につなぐ
+      tagRepo.updateTag(id, { name });
+
+      updateApiTags();
+      updateView();
+    };
+    const onCreditFilterDeleteTag = (id: string) => {
+      // TODO: api につなぐ
+      tagRepo.deleteTag(id);
+      courseRepo.deleteTag(id);
+
+      const tags = creditTags
+        .filter((tag) => tag.id !== id)
+        .map((tag, i) => ({ id: tag.id, order: i }));
+
+      // TODO: api につなぐ
+      tagRepo.changeOrders(tags);
+
+      updateApiTags();
+      updateView();
+    };
+
+    /** credit-course-list-content */
+    const creditCourseWithStateList: CreditCourseWithState[] = reactive(
+      apiToCreditCourseWithStateList(apiCourses.value, apiTags.value)
+    );
+    const displayCreditCourseWithStateList = computed(() => {
+      return selectedTagId.value == undefined
+        ? creditCourseWithStateList
+        : creditCourseWithStateList.filter((ccws) => {
+            const tag = ccws.tags.find((tag) => tag.id === selectedTagId.value);
+            return tag != undefined && tag.assign;
+          });
+    });
+
+    const onCreditCourseListContentCreateTag = (
+      ccws: CreditCourseWithState,
+      tagName: string
+    ) => {
+      // TODO: api につなぐ
+      const newTag = tagRepo.createTag({
+        name: tagName,
+        order: creditTags.length,
+      });
+
+      const assignedTagIds = ccws.tags
+        .filter((tag) => tag.assign)
+        .map(({ id }) => id);
+      assignedTagIds.push(newTag.id);
+
+      // TODO: api につなぐ
+      courseRepo.updateTags(ccws.id, assignedTagIds);
+
+      updateApiCourses();
+      updateApiTags();
+      updateView();
+    };
+
+    const onCreditCourseListContentClickTag = (
+      ccws: CreditCourseWithState,
+      clickedTag: DisplayTag
+    ) => {
+      if (mode.value === "edit") return;
+      const assignedTagIds = ccws.tags
+        .filter((tag) => tag.id !== clickedTag.id && tag.assign)
+        .map(({ id }) => id);
+      if (!clickedTag.assign) assignedTagIds.push(clickedTag.id);
+
+      // TODO: api につなぐ
+      courseRepo.updateTags(ccws.id, assignedTagIds);
+
+      updateApiCourses();
+      updateView();
+    };
+
+    watch(
+      () => mode.value,
```

Comment:
refはそのまま引数におけるはず
https://v3.ja.vuejs.org/api/computed-watch-api.html#watch

```suggestion
      mode,
```